### PR TITLE
Add input validators & logs

### PR DESCRIPTION
**Pull request by zaelgohary**
add input validators & logs in Presearch solution

### CODE REVIEW

File: src/weblets/tf_presearch.vue [hunk 1]
```diff
@@ -13,59 +13,94 @@
...
+          <v-textarea label="Public Presearch Restore Key" v-model="publicRestoreKey" no-resize :spellcheck="false" />
+        </template>
+  
+      </d-tabs>
+    </template>
     <template #footer-actions>
-      <v-btn color="primary" variant="tonal" :loading="loading" :disabled="loading" @click="deploy">
+      <v-btn color="primary" variant="tonal" :loading="loading" :disabled="isInvalid" @click="deploy">
         Deploy
       </v-btn>
```
Comment (MohamedElmdary) [inline src/weblets/tf_presearch.vue]:
```suggestion
      <v-btn color="primary" variant="tonal" :disabled="isInvalid" @click="deploy">
        Deploy
      </v-btn>
```

File: src/weblets/tf_presearch.vue [hunk 2]
```diff
@@ -78,12 +113,15 @@ const farm = ref() as Ref<Farm>
 const loading = ref(false)
```
Comment (MohamedElmdary) [inline src/weblets/tf_presearch.vue]:
```suggestion
```

File: src/weblets/tf_presearch.vue [hunk 3]
```diff
@@ -78,12 +113,15 @@ const farm = ref() as Ref<Farm>
...
 const privateRestoreKey = ref("") as Ref<string>
 const publicRestoreKey = ref("") as Ref<string>
-
+const isBaseValid = ref(false)
+const isInvalid = computed(() => !isBaseValid.value)
+const country = ref<string>()
 
 async function deploy() {
   loading.value = true
```
Comment (MohamedElmdary) [inline src/weblets/tf_presearch.vue]:
```suggestion
```

File: src/weblets/tf_presearch.vue (hunk 1 above)
Comment (zaelgohary) [reply src/weblets/tf_presearch.vue]: removed.

**Review by MohamedElmdary — APPROVED** [review]

**Pull request merged**